plugins/base/src/main/kotlin/signatures/KotlinSignatureProvider.kt

```diff
@@ -103,7 +103,7 @@ class KotlinSignatureProvider(ctcc: CommentsToContentConverter, logger: DokkaLog
     }
 
     private fun <T : DClasslike> classlikeSignature(c: T): List<ContentNode> {
-        @Suppress("UNCHECKED_CAST")
+        @Suppress("UNCHECKED_CAST", "TYPE_MISMATCH_WARNING_FOR_INCORRECT_CAPTURE_APPROXIMATION")
```

Comment:
More info on `TYPE_MISMATCH_WARNING_FOR_INCORRECT_CAPTURE_APPROXIMATION`: [KT-49404](https://youtrack.jetbrains.com/issue/KT-49404/Fix-type-unsoundness-for-contravariant-captured-type-based-on-Java-class)

This will be reported as error in 2.0, so we need to fix it either now or in 1.9. And I'm not sure if fixing it would result in breaking API changes. If so, it's quite sad as this is a frequently used function, and in this case I think we'd better do it sooner than later.

So weed to fix the warning for the `get()` operator override, and then remove all existing `TYPE_MISMATCH_WARNING_FOR_INCORRECT_CAPTURE_APPROXIMATION` suppresses in this PR